aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/BaseHandlerStd.java

```diff
@@ -219,21 +228,31 @@ public abstract class BaseHandlerStd extends BaseHandler<CallbackContext> {
             .backoff(Constant.of().delay(Duration.ofSeconds(30)).timeout(Duration.ofMinutes(180)).build())
             .build();
 
-    private final FilteredJsonPrinter PARAMETERS_FILTER = new FilteredJsonPrinter("MasterUsername", "MasterUserPassword", "TdeCredentialPassword");
+    protected static final RuntimeException MISSING_METHOD_VERSION_EXCEPTION = new RuntimeException("Missing method version");
+
+    protected final HandlerConfig config;
+
+    private final ApiVersionDispatcher<ResourceModel, CallbackContext> apiVersionDispatcher;
 
-    protected HandlerConfig config;
+    private final FilteredJsonPrinter PARAMETERS_FILTER = new FilteredJsonPrinter("MasterUsername", "MasterUserPassword", "TdeCredentialPassword");
 
     public BaseHandlerStd(final HandlerConfig config) {
         super();
         this.config = config;
+        this.apiVersionDispatcher = new ApiVersionDispatcher<ResourceModel, CallbackContext>()
+                .register(ApiVersion.V12, (m, c) -> !software.amazon.awssdk.utils.CollectionUtils.isNullOrEmpty(m.getDBSecurityGroups()));
```

Comment:
Are there any other deprecated attributes are going to force us to use an old version of API?

Reply:
This is unclear by now, but there is a chance.